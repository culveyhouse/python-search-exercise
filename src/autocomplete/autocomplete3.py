class Autocomplete3:
    def __init__(self, suggestions):
        self.suggestions = suggestions

    def performSearch(self, term):
        """Searches through the possible suggestions, creating an array of
        matches for any books whose title matches the specified search term.

        This should perform a case insensitive contains search and sort the
        results by title.

        Example:
            With 'BIOLOGY' specified as the term a book with the title
            'Biology' should be returned. A book with the title 'Microbiology'
            should also be returned.

        Args:
            term - the term to search for in book titles.

        Return:
            List of results where the title contains the search term. Results
            should be sorted alphabetically by title.
        """
        # TODO: Implement this.
        return []
