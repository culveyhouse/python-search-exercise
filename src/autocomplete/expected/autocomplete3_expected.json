[
    {
        "attribution": "Michael Lieberman PhD, Rick Ricer MD",
        "briefDescription": "",
        "chapterPrice": 5.99,
        "copyrightYear": 2013,
        "description": "Studying biochemistry, molecular biology, and genetics? Try the Inkling version! On Inkling, Lieberman and Ricer's BRS Biochemistry, Molecular Biology, and Genetics delivers critical concepts and clinical correlates with interactive slideshows and diagrams that let you test yourself. Plus, take a quiz of USMLE-style review questions with answer feedback so you'll ace your exams, no sweat.",
        "edition": "6th edition",
        "price": 46.99,
        "primaryCategory": "Medicine",
        "publisher": "Lippincott Williams & Wilkins",
        "subCategories": [],
        "subtitle": "",
        "thumbnail": "data/thumbnails/book-01.jpeg",
        "title": "BRS Biochemistry, Molecular Biology, and Genetics",
        "url": "https://www.inkling.com/store/book/brs-biochemistry-molecular-biology-genetics-lieberman-6th/"
    },
    {
        "attribution": "Louise Hawley et al.",
        "briefDescription": "",
        "chapterPrice": 5.99,
        "copyrightYear": 2013,
        "description": "Studying microbiology and immunology? Try the Inkling version! On Inkling, BRS Microbiology & Immunology delivers critical concepts in immunological function and human infection with interactive slideshows and diagrams that let you test yourself. Plus, take a quiz of USMLE-style review questions with answer feedback so you'll ace your exams, no sweat.",
        "edition": "6th edition",
        "price": 46.99,
        "primaryCategory": "Medicine",
        "publisher": "Lippincott Williams & Wilkins",
        "subCategories": [],
        "subtitle": "",
        "thumbnail": "data/thumbnails/book-01.jpeg",
        "title": "BRS Microbiology & Immunology",
        "url": "https://www.inkling.com/store/book/brs-microbiology-immunology-hawley-6th/"
    }
]
